crux-core/src/crux/codec.clj

```diff
@@ -274,6 +274,10 @@
         (id->buffer id to)
         (throw (IllegalArgumentException. (format "Not a %s hex, keyword, URL or an UUID string: %s" hash/id-hash-algorithm this))))))
 
+  PersistentArrayMap
```

Comment:
@t-taylor - reckon this could do with being `java.util.Map`, as discussed (would also mean removing the `IPersistentMap` implementation below.

Also, would it be worth us running with `(into (sorted-map) this)` instead? not convinced normal `{}` _has_ to preserve ordering...

Reply:
Do we want to always alter the map? The only reason we need to touch the arraymap is to remove its ordered quality, but all the other types of maps don't need to be altered. Making it so we always alter the map would effect ingest a little but i agree it would make the code cleaner. Also it should probably be `(into (hash-map) this)` because that 100% doesn't preserve order.

Reply:
I'd rather go the other way and ensure that maps are always ordered consistently - there's nothing in the contract of Map that mandates it returns the same order every time, unless you explicitly use a sorted map. even in Java-land, there's a whole load of Map implementations that might return a different ordering.

(in practice, the same Map concrete class will _usually_ return the elements in the same order, but this isn't something I'd want to rely on, given it's not in the contract, and given the user might use different concrete classes over time)

Reply:
okay that make sense. I think I still disagree because I fell we should do the least amount of work to provide a consistent hash, and having to sort is unnecessary because of the definition of a hash map, i.e. it has no order so the hash of the data structure should only depend on elements rather than order, but I'll concede because i think its just the haskeller in me trying to find the least restrictive function.

Reply:
> the hash of the data structure should only depend on elements rather than order
agreed - we're not using hashes of the object, though - we're using the hash of the serialised form of the map, which doesn't have the same guarantees?